tiledb/sm/array_schema/enumeration.cc

```diff
@@ -191,11 +196,36 @@ Enumeration::Enumeration(
     throw_if_not_ok(offsets_.write(offsets, 0, offsets_size));
   }
 
-  generate_value_map();
+  if (async) {
+    tiledb::common::ThreadPool::Task value_map_future =
+        resources.compute_tp().async(
+            [](std::reference_wrapper<Enumeration> enumeration) {
+              enumeration.get().generate_value_map();
+              return Status::Ok();
+            },
+            std::reference_wrapper<Enumeration>(*this));
+    value_map_future_ = std::move(value_map_future);
+  } else {
+    generate_value_map();
+    value_map_status_.emplace(Status::Ok());
+  }
+}
+
+Enumeration::~Enumeration() {
+  if (value_map_future_.valid()) {
+    auto st = value_map_future_.wait();
+    if (!st.ok()) {
+      tiledb::common::global_logger().warn(
+          "Enumeration::~Enumeration(): Error computing value map: " +
+          st.to_string());
+    }
+  }
 }
```

Comment:
I wouldn't log the warning; it's not important to tell the user that an operation they did not care for failed.